components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/UserSessionManagementService.java

```diff
@@ -28,11 +32,44 @@
     /**
      * Terminates all active sessions of the given user.
      *
-     * @param username username
+     * @param username        username
      * @param userStoreDomain userstore domain of the user
-     * @param tenantDomain tenant domain of the user
+     * @param tenantDomain    tenant domain of the user
      * @throws UserSessionException
      */
     void terminateSessionsOfUser(String username, String userStoreDomain, String tenantDomain) throws
             UserSessionException;
+
+    /**
+     * Get all the active sessions of the given user id.
+     *
+     * @param userId unique id of the user
+     * @throws SessionManagementException if the session retrieval fails
+     */
+    default List<UserSession> getSessionsByUserId(String userId) throws SessionManagementException {
```

Comment:
Shall we @return details. Here and the other 2 methods?

Reply:
Done